TestGetListNonRecursive is flaky

Example failure:
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/116172/pull-kubernetes-unit/1636257857106087936

```
{Failed  === RUN   TestGetListNonRecursive/existing_key,_resourceVersion=0,_resourceVersionMatch=notOlderThan
=== PAUSE TestGetListNonRecursive/existing_key,_resourceVersion=0,_resourceVersionMatch=notOlderThan
=== CONT  TestGetListNonRecursive/existing_key,_resourceVersion=0,_resourceVersionMatch=notOlderThan
    store_tests.go:1421: incorrect list pods: differs from all items, with first:   []example.Pod{
        + 	{
        + 		ObjectMeta: v1.ObjectMeta{Name: "foo", Namespace: "test-ns", ResourceVersion: "2"},
        + 	},
          }
    --- FAIL: TestGetListNonRecursive/existing_key,_resourceVersion=0,_resourceVersionMatch=notOlderThan (0.00s)
```

/kind flake
/sig api-machinery

/assign

--- a/staging/src/k8s.io/apiserver/pkg/storage/testing/store_tests.go
+++ b/staging/src/k8s.io/apiserver/pkg/storage/testing/store_tests.go
@@ -1304,13 +1304,13 @@ func RunTestGetListNonRecursive(ctx context.Context, t *testing.T, store storage
 		name:                 "existing key, resourceVersion=0",
 		key:                  key,
 		pred:                 storage.Everything,
-		expectedAlternatives: [][]example.Pod{{}, {*storedObj}},
+		expectedAlternatives: [][]example.Pod{{}, {*prevStoredObj}, {*storedObj}},
 		rv:                   "0",
 	}, {
 		name:                 "existing key, resourceVersion=0, resourceVersionMatch=notOlderThan",
 		key:                  key,
 		pred:                 storage.Everything,
-		expectedAlternatives: [][]example.Pod{{}, {*storedObj}},
+		expectedAlternatives: [][]example.Pod{{}, {*prevStoredObj}, {*storedObj}},
 		rv:                   "0",
 		rvMatch:              metav1.ResourceVersionMatchNotOlderThan,
 	}, {

